src/sys/unix.rs

```diff
@@ -942,11 +937,19 @@ impl crate::Socket {
         feature = "all",
         any(target_os = "android", target_os = "fuchsia", target_os = "linux")
     ))]
-    pub fn device(&self) -> io::Result<Option<CString>> {
+    pub fn device(&self) -> io::Result<Option<Vec<u8>>> {
+        use std::convert::TryInto as _;
+
+        // TODO(https://github.com/rust-lang/libc/pull/2010): Remove this.
+        #[cfg(target_os = "fuchsia")]
+        const IFNAMSIZ: libc::size_t = 16;
```

Comment:
Let's update libc (after https://github.com/rust-lang/libc/pull/2013 is merged) and use those constants.

Socket2 used to be filled with defined constants from libc, I don't want to go back to that.

Reply:
Pr to update libc: https://github.com/rust-lang/libc/pull/2014.